resources/model/OpenStudio.idd

```diff
@@ -38426,6 +38426,83 @@ OS:ExternalInterface:FunctionalMockupUnitExport:To:Variable,
 
 \group User Defined HVAC and Plant Component Models
 
+OS:Coil:UserDefined,
+       \memo Defines a generic air system component for custom modeling
+       \memo using Energy Management System or External Interface
+       \min-fields 5
+  A1, \field Handle
+       \type handle
+       \required-field        
+  A2, \field Name
+       \required-field
+       \type alpha
+       \reference UserDefinedCoil
+       \reference ConnectionObject
+  A3, \field Overall Model Simulation Program Calling Manager Name
+       \type object-list
+       \object-list ErlProgramCallingManagerNames
+  A4, \field Model Setup and Sizing Program Calling Manager Name
+       \type object-list
+       \object-list ErlProgramCallingManagerNames
+  N1, \field Number of Air Connections
+       \type integer
+       \minimum 1
+       \maximum 2
+       \required-field
+  A5, \field Air Connection 1 Inlet Node Name
+       \type object-list
+       \object-list ConnectionNames
+       \required-field
+  A6, \field Air Connection 1 Outlet Node Name
+       \type object-list
+       \object-list ConnectionNames
+       \required-field
+  A7, \field Plant Connection is Used
+       \type choice
+       \default No
```

Comment:
Generally we remove "default" make these "required-field" (and then explicitly set the default value in the ctor). Not totally sure if that should happen here. One benefit is that you'd then no longer need `xxxIsDefaulted` or `resetXXX` methods (looks like you don't have these anyway...)

Reply:
yeah, i was considering removing that from the idd since E+ is the only one who needs it and its set in the FT https://github.com/NREL/OpenStudio/blob/4449-coil/src/energyplus/ForwardTranslator/ForwardTranslateCoilUserDefined.cpp#L78